core/src/main/java/org/bitcoinj/core/StoredBlock.java

```diff
@@ -129,7 +129,10 @@ public StoredBlock build(Block block) throws VerificationException {
      * @return the previous block in the chain or null if it was not found in the store.
      */
     public StoredBlock getPrev(BlockStore store) throws BlockStoreException {
-        return store.get(getHeader().getPrevBlockHash());
+        if (height == 0)
+            // Genesis blocks have no previous block.
+            return null;
+        return store.get(header.getPrevBlockHash());
```

Comment:
I would suggest using a single `return` statement and a ternary operator here.

Reply:
Changed.